xonsh/tools.py

```diff
@@ -92,6 +93,85 @@ def __init__(self, returncode, command, output=None, stderr=None,
         self.completed_command = completed_command
 
 
+def expandpath(path):
+    """
+    Performs environment variable / user expansion on a given path
+    if the relevant flag has been set.
+    """
+    env = getattr(builtins, '__xonsh_env__', os.environ)
+    if env.get('EXPAND_ENV_VARS', False):
+        # expand variables and use os.path.realpath to handle cases
+        # with relative paths like ../ or ./
+        path = os.path.expanduser(expandvars(path))
+    return os.path.realpath(path)
```

Comment:
I don't think we should call realpath() here.  If there is a symbolic link on the path it is OK.